mypy: `error: invalid syntax  [syntax]` at `frictionless/dialect/types.py:28`

Dear @roll,

first things first: Thank you so much for conceiving and maintaining Frictionless. It is true gold.

At https://github.com/daq-tools/skeem/pull/10, we believe we are being tripped by a similar issue already reported at GH-1580 by @aborruso.

At our end, it is the mypy type checker, which fails like:
```
$ mypy skeem
.venv/lib/python3.10/site-packages/frictionless/dialect/types.py:28: error: invalid syntax  [syntax]
Found 1 error in 1 file (errors prevented further checking)
```
... when upgrading to frictionless-5.14.x already:
```
pip install --upgrade "frictionless<5.15"
```

On the other hand, when downgrading to frictionless-5.13.x:
```
pip install --upgrade "frictionless<5.14"
```

... mypy succeeds again:
```
mypy --install-types --non-interactive skeem
Success: no issues found in 36 source files
```

With kind regards,
Andreas.

P.S.: We are using the most recent mypy 1.5.1:
```
$ mypy --version
mypy 1.5.1 (compiled: yes)
```

> https://github.com/python/mypy/issues/10768 indicates that this invocation and error report may mask the original error.

Indeed. I only received the error on a fresh checkout, i.e. without an existing `.mypy_cache` directory. When it is present (`mkdir .mypy_cache`), the error goes back to:
```
site-packages/frictionless/resource/types.py:14: error: invalid syntax  [syntax]
```
... as outlined above.

--- a/frictionless/resource/types.py
+++ b/frictionless/resource/types.py
@@ -11,7 +11,6 @@ from ..schema import ISchema
 # TODO: replace by frictionless-standards
 class IResource(TypedDict, total=False):
     name: Required[str]
-    #  type = required
     title: str
     description: str
     path: str

--- a/frictionless/schema/types.py
+++ b/frictionless/schema/types.py
@@ -18,7 +18,6 @@ class ISchema(TypedDict, total=False):
 
 class IField(TypedDict, total=False):
     name: Required[str]
-    #  type = required
     title: str
     description: str
     format: str

